test/os/filter/impl/ecql/exclusionformatter.test.js

```diff
@@ -0,0 +1,33 @@
+goog.require('os.mock');
+goog.require('ol.Feature');
+goog.require('ol.geom.Polygon');
+goog.require('os.filter.impl.ecql.ExclusionFormatter');
+
+describe('os.filter.impl.ecql.ExclusionFormatter', function() {
+  it('should format geometries', function() {
+    var f = new ol.Feature(ol.geom.Polygon.fromExtent([0, 1, 2, 3]));
+    var formatter = new os.filter.impl.ecql.ExclusionFormatter();
+    expect(formatter.format(f)).toBe('(DISJOINT(geometry,POLYGON((0 1,0 3,2 3,2 1,0 1))))');
+  });
+
+  it('should format geometries with other column names', function() {
+    var f = new ol.Feature(ol.geom.Polygon.fromExtent([0, 1, 2, 3]));
+    var formatter = new os.filter.impl.ecql.ExclusionFormatter('other');
+    expect(formatter.format(f)).toBe('(DISJOINT(other,POLYGON((0 1,0 3,2 3,2 1,0 1))))');
+  });
+
+  it('should wrap multiple geometries properly', function() {
+    var formatter = new os.filter.impl.ecql.ExclusionFormatter();
+    var f = new ol.Feature(ol.geom.Polygon.fromExtent([0, 1, 2, 3]));
+    var result = formatter.format(f);
+    var f2 = new ol.Feature(ol.geom.Polygon.fromExtent([4, 5, 6, 7]));
+    result += formatter.format(f2);
+
+    expect(formatter.wrapMultiple(result)).toBe(
+      '(' +
+      '(DISJOINT(geometry,POLYGON((0 1,0 3,2 3,2 1,0 1))))' +
+      ' OR ' +
```

Comment:
If I'm reading this right, if we have a geometry within one of the exclusion areas (say `POINT(1.5 2.5)` which is within `f`) then this filter will return true.

There is probably no right answer (users being users...) but I would have assumed that given two exclusion areas, it would be appropriate to exclude anything that intersects either of them. Right now, no points will be excluded (because no point can be disjoint to both).

At least I think so - this double negative stuff is doing my head in..

Reply:
That's not really that function's concern. If there's a bug with the exclusion condition, then it is in `os.ui.filter.QueryHandler.prototype.wrap()`.

Your confusion is why I wrote #209.